### PR TITLE
Add CRD for ActionApproval

### PR DESCRIPTION
**Pull request by harsh-px**
- remove approval fields from AutopilotRuleobject

This CRD has following benefits

- Allows multiple action approval workflows per object. Previously, a single CRD object had a list of approvals.
- Decouples approval from the AutopilotRuleObject whose main purpose was to just track status of all events per object
- Conforms to the multi-tenancy model as it's a namespaced resource

Signed-off-by: Harsh Desai <harsh@portworx.com>

### CODE REVIEW

File: pkg/apis/autopilot/v1alpha1/actionapproval.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	meta.TypeMeta   `json:",inline"`
+	meta.ObjectMeta `json:"metadata,omitempty"`
+	Spec            ActionApprovalSpec   `json:"spec"`
+	Status          ActionApprovalStatus `json:"status"`
+}
+
+// ActionApprovalSpec is the spec for an autopilot action approval
+type ActionApprovalSpec struct {
+	// ApprovalState is the current approval state of the action
+	ApprovalState ActionApprovalState `json:"approvalState,omitempty"`
```
Comment (adityadani) [inline pkg/apis/autopilot/v1alpha1/actionapproval.go:38]: nit: can you move the ActionApprovalState definitions to this file as well ?

File: pkg/apis/autopilot/v1alpha1/actionapproval.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	// ActionPreviews provides a dry-run preview of the side-effects of the actions
+	ActionPreviews []*AutopilotActionPreview `json:"actionPreviews,omitempty"`
+}
+
+// AutopilotActionPreview is a preview of an action and the expected result of it before it gets executed
+type AutopilotActionPreview struct {
+	// Action is the action spec that's going to get executed
+	Action *RuleAction `json:"action,omitempty"`
+	// ExpectedResult is a user friendly description of the outcome of executing the action
+	ExpectedResult string `json:"expectedResult,omitempty"`
```
Comment (adityadani) [inline pkg/apis/autopilot/v1alpha1/actionapproval.go]: Should we make this ExpectedResult a struct and then add a Message as a string. In this way it is more extensible if we need to have well defined Results.

**Review by adityadani — APPROVED** [review]

File: pkg/apis/autopilot/v1alpha1/actionapproval.go (hunk 1 above)
Comment (harsh-px) [reply pkg/apis/autopilot/v1alpha1/actionapproval.go:38]: done

File: pkg/apis/autopilot/v1alpha1/actionapproval.go (hunk 2 above)
Comment (harsh-px) [reply pkg/apis/autopilot/v1alpha1/actionapproval.go]: Good idea. Done. Also added a String() function which currently just returns the Message. If we add fields in future, that can pretty format the result.

**Pull request merged**